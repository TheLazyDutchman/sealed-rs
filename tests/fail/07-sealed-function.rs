pub mod inner {
    use sealed::sealed;

    #[sealed]
    pub trait PartialSealed {
        #[seal(uncallable)]
        fn a();

        fn b();
    }

    pub struct A;

    #[sealed]
    impl PartialSealed for A {
        #[seal(uncallable)]
        fn a() {}

        fn b() {}
    }
}

use crate::inner::PartialSealed;

fn main() {
    inner::A::a();
    inner::A::a(Token);
    inner::A::b();
}
