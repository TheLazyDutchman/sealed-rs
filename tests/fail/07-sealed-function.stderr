error[E0425]: cannot find value `Token` in this scope
  --> tests/fail/07-sealed-function.rs:27:14
   |
27 |     inner::A::a(Token);
   |                 ^^^^^ not found in this scope
   |
note: unit struct `crate::inner::__seal_partial_sealed::Token` exists but is inaccessible
  --> tests/fail/07-sealed-function.rs:4:2
   |
4  |     #[sealed]
   |     ^^^^^^^^^ not accessible
   = note: this error originates in the attribute macro `sealed` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0061]: this function takes 1 argument but 0 arguments were supplied
  --> tests/fail/07-sealed-function.rs:26:2
   |
26 |     inner::A::a();
   |     ^^^^^^^^^^^-- an argument of type `Token` is missing
   |
note: associated function defined here
  --> tests/fail/07-sealed-function.rs:7:6
   |
7  |         fn a();
   |            ^
help: provide the argument
   |
26 |     inner::A::a(/* Token */);
   |                ~~~~~~~~~~~~~
